### PR TITLE
[NRL-760] Deployment changes for blue/green arch

### PR DESCRIPTION
**Pull request by mattdean3-nhs**

### CODE REVIEW

File: .github/workflows/persistent-environment.yml [hunk 1]
```diff
@@ -180,11 +189,152 @@ jobs:
...
+
+      - name: Setup Python environment
+        run: |
+          poetry install --no-root
+          source $(poetry env info --path)/bin/activate
+
+      - name: Deactivate Stack
+        run: |
+          inactive_stack_name=$(poetry run python ./scripts/get_env_config.py inactive-stack ${{ inputs.environment }})
+          poetry run python ./scripts/activate-stack.py ${inactive_stack_name}
```
Comment (mattdean3-nhs) [inline .github/workflows/persistent-environment.yml:340]: add inputs.environment as param to end of this line

File: scripts/activate_stack.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    environment_config = json.loads(response["SecretString"])
+    print(f"Got environment config for {env}: {environment_config}")
+
+    lock_state = environment_config[CONFIG_LOCK_STATE]
+    if lock_state != "open":
+        print(
+            f"Unable to activate stack as lock state is not open: {lock_state}",
+            file=sys.stderr,
+        )
+        return
```
Comment (mattdean3-nhs) [inline scripts/activate_stack.py:102]: return non-zero exit here

File: scripts/activate_stack.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    current_active_stack = environment_config[CONFIG_ACTIVE_STACK]
+    if current_active_stack == stack_name:
+        print("Cannot activate stack, stack is already active", file=sys.stderr)
+        return
```
Comment (mattdean3-nhs) [inline scripts/activate_stack.py:107]: return non-zero exit here

File: .github/workflows/persistent-environment.yml (hunk 1 above)
Comment (mattdean3-nhs) [reply .github/workflows/persistent-environment.yml:340]: Going to do in next PR

File: scripts/activate_stack.py (hunk 2 above)
Comment (mattdean3-nhs) [reply scripts/activate_stack.py:102]: Going to do in next PR

File: scripts/activate_stack.py (hunk 3 above)
Comment (mattdean3-nhs) [reply scripts/activate_stack.py:107]: Going to do in next PR

**Review by axelkrastek1-nhs — APPROVED** [review]

**Pull request merged**